packages/components/pager/src/pager.js

```diff
@@ -0,0 +1,199 @@
+			});
+		}
+
+		get prev() {
+			return this.navigationButton({
+				className: 'prev-page',
+				disabled: this.activePage === 1,
+				tooltip: this.translations.first_page_tooltip,
+				icon: html`
+					<ts-icon icon="arrow-down-short" size="large" rotate="90"></ts-icon>
+				`,
+				callback: () => this.activePage--
+			});
+		}
+
+		get next() {
+			return this.navigationButton({
+				className: 'prev-page',
+				disabled: this.activePage === this.totalPages,
+				tooltip: this.translations.last_page_tooltip,
+				icon: html`
+					<ts-icon icon="arrow-down-short" size="large" rotate="270"></ts-icon>
+				`,
+				callback: () => this.activePage++
+			});
+		}
+
+		get pageInput() {
+			return html`
+				<li class="page-input-wrapper">
+					<label for="activPage">${this.translations.page}</label>
+					<input
+						id="activPage"
+						class="page-input"
+						type="number"
+						min="1"
+						max="${this.totalPages}"
+						.value="${this.activePage}"
+						@change="${this.handleInputChange}"
+					/>
+					<span>${this.translations.of} ${this.totalPages}</span>
+				</li>
+			`;
+		}
+
+		get itemsPerPageSelctor() {
+			if (!this.perPage) {
+				return;
+			}
+			if (perPageSelectValues.indexOf(this.perPage) === -1) {
+				const errorMessage = `Wrong items per page value: ${
+					this.perPage
+				}. per-page should be either of these ${perPageSelectValues.join()}`;
+				throw new Error(errorMessage);
+			}
+
+			return html`
+				<label class="item-per-page-container">
+					${this.translations.items_per_page}
+					<select id="itemsPerPage" @change="${this.handlePerPageChange}">
+						${perPageSelectValues.map(
+							value =>
+								html`
+									<option value="${value}" ?selected="${value === this.perPage}">${value}</option>
+								`
+						)}
+					</select>
+				</label>
+			`;
+		}
+
+		navigationButton({ className, disabled, tooltip, icon, callback }) {
+			return html`
+				<li class="${className}">
+					<ts-tooltip ?disabled="${!disabled}" tooltip="${tooltip}" Position="top">
+						<button ?disabled=${disabled} @click="${callback}">
+							${icon}
+						</button>
+					</ts-tooltip>
+				</li>
+			`;
+		}
+
+		handleInputChange(e) {
+			let newPageNumber = Number(e.target.value);
+			// blur the input to stop changing the page with arrow multiple times in a row
+			e.target.blur();
+			// if the input was empty, don't change the page
+			if (!e.target.value) {
+				newPageNumber = this.activePage;
+			}
+			if (newPageNumber < 1) {
+				newPageNumber = 1;
+			} else if (newPageNumber > this.totalPages) {
+				newPageNumber = this.totalPages;
+			}
+			this.activePage = newPageNumber;
+			// for those cases changing the input to out of range two time in a row
+			e.target.value = newPageNumber;
+		}
+
+		attributeChangedCallback(name, oldVal, newVal) {
+			super.attributeChangedCallback(name, oldVal, newVal);
+			if (name === 'active-page') {
+				// TODO: make this reusable
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ts.sv/coding_rules#rule_key=javascript%3AS1135)